conans/test/functional/cache/short_paths_test.py

```diff
@@ -132,3 +137,28 @@ class TestConan(ConanFile):
         client.run("install test/1.0@", assert_error=True)
         self.assertIn("ERROR: Missing binary: test/1.0:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9",
                       client.out)
+
+    @parameterized.expand([(True,), (False,)])
+    def test_leaking_folders(self, use_always_short_paths):
```

Comment:
I think we are mixing two different tests here. One is to check that the shortened folders are removed too, and the other one is to check that ``general.use_always_short_paths`` and attribute ``short_paths`` have the same effects. In this PR I would expect only the first one, the other should be already tested.

Reply:
The thing is that this bite me during the testing, so it is clearly not fully tested elsewhere. Actually ``use_always_short_paths`` is very undertested, only some test is checking the env-var, not the conan.conf. So I think it doesn't harm (it is actually good) to test that neither way of defining short-paths leak folders in the short-path storage